src/Uno.Toolkit.UI/Extensions/DependencyObjectExtensions.cs

```diff
@@ -172,5 +176,45 @@ internal static void SetParent(this DependencyObject dependencyObject, object? p
 			}
 		}
 #endif
+
+		public static DependencyProperty? FindDependencyPropertyUsingReflection<TProperty>(this DependencyObject dependencyObject, string propertyName)
+		{
+			var type = dependencyObject.GetType();
+			var propertyType = typeof(TProperty);
+			var key = (ownerType: type, propertyName);
+
+			if (_dependencyPropertyReflectionCache.TryGetValue(key, out var property))
+			{
+				return property;
+			}
+
+			property =
+				type
+					.GetTypeInfo()
+					.GetDeclaredProperty(propertyName)
+					?.GetValue(null) as DependencyProperty
+				?? type
+					.GetTypeInfo()
+					.GetDeclaredField(propertyName)
+					?.GetValue(null) as DependencyProperty;
```

Comment:
It would be nice that the property is static to avoid exception in `GetValue`:

```suggestion
			property =
				type
					.GetProperty(propertyName, BindingFlags.Public | BindingFlags.Static)
					?.GetValue(null) as DependencyProperty
				?? type
					.GetField(propertyName, BindingFlags.Public | BindingFlags.Static)
					?.GetValue(null) as DependencyProperty;
```